app/src/androidTest/java/edu/uc/kovaciad/slicetracker/FirebaseSaveTest.kt

```diff
@@ -8,40 +8,40 @@ import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 
-//@RunWith(AndroidJUnit4::class)
-//class FirebaseSaveTest {
-//    private lateinit var viewModel: MainViewModel
-//
-//    @Before
-//    fun getViewModel() {
-//        viewModel = MainViewModel()
-//    }
-//
-//
-//    @Test
-//    fun savePrinter() {
-//        val printerTest = Printer("Ender 3 Pro", "bhhjf32", "jh34jk423")
-//        val result = viewModel.save(printerTest, "printers")
-//        Log.d("TEST", result.toString())
-//        assert(result)
-//        val printerResult = getPrinter(printerTest.id)
-//        assert(printerResult)
-//    }
-//
-//    private fun getPrinter(id: String): Boolean {
-//        val docRef = viewModel.firestore.collection("printers").document(id)
-//        var printerName = ""
-//        docRef.get()
-//            .addOnSuccessListener { document ->
-//                run {
-//                    if (document != null) {
-//                        printerName = document.toObject(Printer::class.java).toString()
-//                    }
-//                }
-//            }
-//            .addOnFailureListener {
-//                return@addOnFailureListener
-//            }
-//        return printerName == "Ender 3 Pro"
-//    }
-//}
\ No newline at end of file
+@RunWith(AndroidJUnit4::class)
```

Comment:
As the professor said in the last review, experimental code could possiblly be given it's own branch in the repository? I also included @Test for your methods here. If they're unfinished it's an effective way to temporarily name them if they aren't used at runtime: https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods

Reply:
Yes... agree on the commented out code part.

@Test indicates that this is a Unit Test method, and has an assert statement.  That's not necessarily the case here.  Instead of annotating with @Test, I'd either fix the functions, or remove them entirely.